evalml/utils/woodwork_utils.py

```diff
@@ -132,3 +132,29 @@ def _convert_numeric_dataset_pandas(X, y):
         )
     y_ww = infer_feature_types(y)
     return X_ww, y_ww
+
+
+def _schema_is_equal(first, other):
+    """Loosely check whether or not the Woodwork schemas are equivalent. This only checks that the string values for the schemas are equal and doesn't take the actual type objects into account.
+
+    Args:
+        first (ww.Schema): The schema of the first woodwork datatable
+        other (ww.Schema): The schema of the second woodwork datatable
+
+    Returns:
+        bool: Whether or not the two schemas are equal
+    """
+    if first.types.index.tolist() != other.types.index.tolist():
+        return False
+    logical = [
+        x if x != "Integer" else "Double"
```

Comment:
After discussion with @freddyaboulton and @angela97lin, we decided to treat Integer and Doubles as the same logical types until @chukarsten's work with `NullableInteger` goes in. At that point, we can revisit this and see what the best way to accommodate that would be.